src/blockdata/transaction.rs

```diff
@@ -407,17 +406,14 @@ impl Transaction {
 
     #[cfg(feature="bitcoinconsensus")]
     /// Verify that this transaction is able to spend some outputs of spent transactions
-    pub fn verify(&self, spent: &HashMap<sha256d::Hash, Transaction>) -> Result<(), script::Error> {
+    pub fn verify<S>(&self, spent: S) -> Result<(), script::Error>
+        where S: Fn(&OutPoint) -> Option<TxOut> {
```

Comment:
Can we use `FnOnce` rather than `Fn`?

Reply:
the function is called in a loop, I think that is against the rule of FnOnce that it should only be called once. I could make it called once only to resolve all input, but that would complicate the implementation, I do not think it is worth it.

Reply:
Oh, oops, I missed the loop. `FnMut` then? You may need to call the closure with `&mut spent` rather than just `spent`.

I don't actually know in what contexts `FnMut` won't work and you'd need `Fn`.

Reply:
While I was thinking about why one would use FnMut I recognized that this is the best place to catch transactions that try to re-use (double spend) the same input. The resolver should update its state so it is not returning the same txout twice. Also added test for this